venvs/tests/test_converge.py

```diff
@@ -261,6 +261,28 @@ def test_specified_link_name(self):
             {"fooBar": self.locator.for_name("a").binary("foo")},
         )
 
+    def test_missing_link_dir(self):
+        self.filesystem.set_contents(
+            self.locator.root.descendant("virtualenvs.toml"), """
+            [virtualenv.a]
+            link = ["foo"]
+            """
+        )
+
+        self.filesystem.remove(self.link_dir)
+        try:
+            self.run_cli(["converge"])
+        finally:
+            try:
+                self.filesystem.create(self.link_dir)
+            except FileExists:
+                pass
```

Comment:
This seems pretty foul.  `self.link_dir` existing is just part of the `CLIMixin` but, for example, `~/.local/bin` may not exist in real usage.  So how should we implement this 'temporary' lack of `self.link_dir` for this test?

Reply:
Maybe just run `filesystem.remove_empty_directory` on it before this test?

Reply:
Well duh, because they are defined on the class I had in my head the setup and tear down were class scoped not method scoped...